google/cloud/bigtable/mutations_batcher.py

```diff
@@ -46,61 +174,221 @@ class MutationsBatcher:
-    conditional_queues: dict[RowFilter, tuple[list[Mutation], list[Mutation]]]
-
-    MB_SIZE = 1024 * 1024
-
     def __init__(
         self,
         table: "Table",
-        flush_count: int = 100,
-        flush_size_bytes: int = 100 * MB_SIZE,
-        max_mutation_bytes: int = 20 * MB_SIZE,
-        flush_interval: int = 5,
-        metadata: list[tuple[str, str]] | None = None,
+        *,
+        flush_interval: float | None = 5,
+        flush_limit_count: int | None = 100,
+        flush_limit_bytes: int | None = 20 * MB_SIZE,
+        flow_control_max_count: int | None = 100000,
+        flow_control_max_bytes: int | None = 100 * MB_SIZE,
     ):
-        raise NotImplementedError
+        """
+        Args:
+          - table: Table to preform rpc calls
+          - flush_interval: Automatically flush every flush_interval seconds
+          - flush_limit_count: Flush immediately after flush_limit_count mutations are added.
+              If None, this limit is ignored.
+          - flush_limit_bytes: Flush immediately after flush_limit_bytes bytes are added.
+              If None, this limit is ignored.
+          - flow_control_max_count: Maximum number of inflight mutations.
+              If None, this limit is ignored.
+          - flow_control_max_bytes: Maximum number of inflight bytes.
+              If None, this limit is ignored.
+        """
+        atexit.register(self._on_exit)
+        self.closed: bool = False
+        self._table = table
+        self._staged_mutations: list[RowMutationEntry] = []
+        self._staged_count, self._staged_bytes = 0, 0
+        self._flow_control = _FlowControl(
+            flow_control_max_count, flow_control_max_bytes
+        )
+        self._flush_limit_bytes = (
+            flush_limit_bytes if flush_limit_bytes is not None else float("inf")
+        )
+        self._flush_limit_count = (
+            flush_limit_count if flush_limit_count is not None else float("inf")
+        )
+        self.exceptions: list[FailedMutationEntryError] = []
+        self._flush_timer_task: asyncio.Task[None] = asyncio.create_task(
+            self._flush_timer(flush_interval)
+        )
+        # create noop previous flush task to avoid None checks
+        self._prev_flush: asyncio.Task[None] = asyncio.create_task(asyncio.sleep(0))
+        # MutationExceptionGroup reports number of successful entries along with failures
+        self._entries_processed_since_last_raise: int = 0
+
+    async def _flush_timer(self, interval: float | None):
+        """
+        Triggers new flush tasks every `interval` seconds
+        """
+        if interval is None:
+            return
+        while not self.closed:
+            await asyncio.sleep(interval)
+            # add new flush task to list
+            if not self.closed and self._staged_mutations:
+                self._schedule_flush()
+
+    def append(self, mutations: RowMutationEntry | list[RowMutationEntry]):
```

Comment:
Lets avoid adding multiple entries at a time. To support this usage we would want to break up the list into batches, which will complicate things.